test.py

```diff
@@ -64,6 +64,26 @@ def test_raises_for_non_csv(self, mock_urlopen):
         with self.assertRaisesRegexp(Exception, err):
             self.stream._api_call(None)
 
+    # Handles unencodable strings
+    @patch('ocr.urllib2.urlopen')
+    def test_hanldes_unencodable_strings(self, mock_urlopen):
+        res_mock = Mock()
+        res_mock.info.side_effect = self.foo
+        # The characters here represent a copywrite symbol
+        res_mock.read.side_effect = lambda: "{'problem_text': 'OHNO!\xc2\xae'}"
+        mock_urlopen.return_value = res_mock
+
+        exception_raised = False
+        try:
+            self.stream.read()
+        except Exception, e:
+            exception_raised = e
+
+        self.assertFalse(exception_raised, exception_raised)
```

Comment:
you are right.. but still. it make more sense to test the returned data by read than see no errors are thrown. also, in case of error without catch test will fail, so it does cover also error case